lib/AST/Decl.cpp

```diff
@@ -3997,14 +3995,19 @@ void NominalTypeDecl::synthesizeSemanticMembersIfNeeded(DeclName member) {
   }
 }
 
+bool ClassDecl::hasCircularInheritance() const {
```

Comment:
Do you expect to call these methods from anywhere else? If not, I think you can just inline them into their caller.

Reply:
@slavapestov I was thinking about using `hasCircularInheritance` to temporarily help reduce some redundant circularity diagnostics until we come up with a more principled solution